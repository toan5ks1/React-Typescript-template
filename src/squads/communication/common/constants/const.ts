import { convertEnumKeys } from "src/common/constants/helper";

import { MediaType, UserGroup, UserNotificationStatus } from "manabie-bob/enum_pb";
import { ConversionTaskStatus } from "manabuf/bob/v1/media_pb";
import {
    NotificationEvent,
    NotificationStatus,
    NotificationTargetGroupSelect,
    NotificationType,
    UserNotificationQuestionnaireStatus,
    QuestionType,
} from "manabuf/common/v1/notifications_pb";

// TODO: Remove and replace the real roles after finish AC Roles phase
// https://manabie.atlassian.net/wiki/spaces/LT/pages/463604245/FE+hardcoded+logic+for+AC+Roles
export enum ExtendedUserGroup {
    USER_GROUP_HQ_STAFF = 16,
    USER_GROUP_TEACHER_LEAD = 17,
    USER_GROUP_CENTRE_LEAD = 18,
    USER_GROUP_CENTRE_MANAGER = 19,
    USER_GROUP_CENTRE_STAFF = 20,
}

export const UserGroupUnion = Object.assign({}, UserGroup, ExtendedUserGroup);

export const UserRoles = convertEnumKeys(UserGroupUnion); // this is used frequently, so I dont change it naming now
export const KeyMediaTypes = convertEnumKeys(MediaType);
export const KeyConversionTaskStatusTypes = convertEnumKeys(ConversionTaskStatus);
export const KeyNotificationType = convertEnumKeys(NotificationType);
export const KeyNotificationTargetGroupSelect = convertEnumKeys(NotificationTargetGroupSelect);
export const KeyNotificationStatus = convertEnumKeys(NotificationStatus);
export const KeyNotificationEvent = convertEnumKeys(NotificationEvent);
export const KeyUserNotificationStatus = convertEnumKeys(UserNotificationStatus);
export const KeyQuestionnaireStatus = convertEnumKeys(UserNotificationQuestionnaireStatus);
export const KeyQuestionTypes = convertEnumKeys(QuestionType);

export const DEFAULT_DEBOUNCE_AUTOCOMPLETE_TIME = 200;
