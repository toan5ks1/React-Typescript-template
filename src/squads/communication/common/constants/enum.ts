export const MAX_LENGTH_EDITOR = 10240;
export const MAX_LENGTH_TITLE = 255;

export const MAX_ANSWER_NUMBER = 10;
export const MIN_ANSWER_NUMBER = 2;

export const MAX_QUESTION_NUMBER = 10;

export const MAX_LENGTH_QUESTION_CONTENT = 4000;
export const MAX_LENGTH_ANSWER_CONTENT = 4000;

export const MAX_LENGTH_TAG = 256;

export enum FileType {
    CSV = "csv",
}
