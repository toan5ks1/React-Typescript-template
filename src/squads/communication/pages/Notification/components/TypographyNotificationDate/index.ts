export { default } from "./TypographyNotificationDate";
export * from "./TypographyNotificationDate";
