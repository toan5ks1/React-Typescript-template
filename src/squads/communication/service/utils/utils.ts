export function createEmptyResponse<T = any>(resp: T): Promise<T> {
    return Promise.resolve(resp);
}

export function getSearchString(text: string | null | undefined) {
    if (!text) return undefined;
    return `%${text}%`;
}
