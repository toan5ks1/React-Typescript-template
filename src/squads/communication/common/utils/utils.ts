import { DateTime, DurationLikeObject } from "luxon";
import {
    KeyNotificationEvent,
    KeyNotificationStatus,
    KeyNotificationType,
    UserRoles,
} from "src/common/constants/const";
import { ArrayElement, FilterAppliedObjectsMap } from "src/common/constants/types";
import { arrayHasItem, pick1stElement } from "src/common/utils/other";
import { combineDateAndTime } from "src/common/utils/time";
import {
    NotificationFormData,
    NotificationStatusType,
    UpsertNotificationProps,
} from "src/squads/communication/common/constants/types";
import { getAlphabetArray } from "src/squads/communication/common/utils/other";
import { createQuestionsListFromQuestionFieldArray } from "src/squads/communication/common/utils/questionnaire-utils";
import { Grade } from "src/squads/communication/models/grade";
import { NotificationSelectors } from "src/squads/communication/models/notification";
import { CoursesManyReferenceQuery } from "src/squads/communication/service/bob/bob-types";

import { NotificationStatus } from "manabuf/common/v1/notifications_pb";
import { Paging } from "manabuf/common/v1/requests_pb";

import { UseResourceTranslateReturn } from "src/squads/communication/hooks/useResourceTranslate";

interface GetQuestionnaireUpsertDataParams
    extends Pick<
        NotificationFormData,
        "expirationDate" | "expirationTime" | "questionFieldArrayItem" | "isAllowResubmission"
    > {}

interface GetScheduleUpsertDataParams
    extends Pick<NotificationFormData, "scheduleDate" | "scheduleTime" | "status"> {}

const isNotificationStatusType = (status: string): status is NotificationStatusType => {
    return Object.keys(KeyNotificationStatus).includes(status);
};

export function getConvertedStatus(
    status: string | (string | null)[] | null
): NotificationStatusType | undefined {
    if (!status || status === null) return undefined;

    if (Array.isArray(status)) {
        const firstNotificationStatus = pick1stElement(status);

        if (firstNotificationStatus && isNotificationStatusType(firstNotificationStatus)) {
            return firstNotificationStatus;
        }
    } else if (isNotificationStatusType(status)) return status;
}

//TODO: @communication remove checkDraftStatus after Notification V2 release and Notification V1 removed
export function checkDraftStatus(status?: string): boolean {
    return status === KeyNotificationStatus.NOTIFICATION_STATUS_DRAFT;
}
//TODO: @communication remove checkScheduleStatus after Notification V2 release and Notification V1 removed
export function checkScheduleStatus(status?: string): boolean {
    return status === KeyNotificationStatus.NOTIFICATION_STATUS_SCHEDULED;
}
//TODO: @communication remove checkSentStatus after Notification V2 release and Notification V1 removed
export function checkSentStatus(status?: string): boolean {
    return status === KeyNotificationStatus.NOTIFICATION_STATUS_SENT;
}

export function checkDraftStatusV2(status?: NotificationStatus): boolean {
    return status === NotificationStatus.NOTIFICATION_STATUS_DRAFT;
}

export function checkSentStatusV2(status?: NotificationStatus): boolean {
    return status === NotificationStatus.NOTIFICATION_STATUS_SENT;
}

export function checkScheduleStatusV2(status?: NotificationStatus): boolean {
    return status === NotificationStatus.NOTIFICATION_STATUS_SCHEDULED;
}

export const toDraftSelector = (selector: string) => {
    return `[data-js="${selector}"] .public-DraftEditor-content`;
};

export function selectInnerHTMLContentField(): { contentHTML: string } {
    const draftSelector = toDraftSelector(NotificationSelectors.content);
    const contentHTML = document.querySelector(draftSelector)?.innerHTML || "";

    return { contentHTML };
}

const getCoursesUpsertData = (courses: NotificationFormData["courses"]) => {
    let courseIds: UpsertNotificationProps["courseIds"] = [];

    const isAllCourses =
        courses.findIndex(
            (course: ArrayElement<CoursesManyReferenceQuery["courses"]>) =>
                typeof course.course_id === "undefined"
        ) >= 0;

    if (!isAllCourses) {
        courseIds = courses
            .filter((course) => Boolean(course.course_id))
            .map((course) => course.course_id);
    }

    return { isAllCourses, courseIds };
};

const getGradesUpsertData = (grades: NotificationFormData["grades"]) => {
    let gradeIds: UpsertNotificationProps["gradeIds"] = [];

    const isAllGrades = grades.findIndex((grade: Grade) => typeof grade.id === "undefined") >= 0;

    if (!isAllGrades) {
        gradeIds = grades.map(({ id }) => Number(id));
    }

    return { isAllGrades, gradeIds };
};

const getQuestionnaireUpsertData = ({
    expirationDate,
    expirationTime,
    questionFieldArrayItem,
    isAllowResubmission,
}: GetQuestionnaireUpsertDataParams) => {
    if (
        expirationDate &&
        expirationTime &&
        expirationTime.value &&
        questionFieldArrayItem &&
        arrayHasItem(questionFieldArrayItem)
    ) {
        const questionsList = createQuestionsListFromQuestionFieldArray(questionFieldArrayItem);

        return {
            questionnaireId: "",
            questionsList,
            resubmitAllowed: Boolean(isAllowResubmission),
            expirationDate: combineDateAndTime(expirationDate, expirationTime.value),
        };
    }
};

const getScheduleUpsertData = ({
    scheduleDate,
    scheduleTime,
    status,
}: GetScheduleUpsertDataParams) => {
    if (
        scheduleDate &&
        scheduleTime &&
        scheduleTime.value &&
        status === KeyNotificationStatus.NOTIFICATION_STATUS_SCHEDULED
    ) {
        return combineDateAndTime(scheduleDate, scheduleTime.value);
    }
};

export const getMappedNotificationUpsertData = (
    formValues: NotificationFormData
): UpsertNotificationProps => {
    const trimmedTitle = formValues.title.trim();

    const mappedContent = {
        raw: formValues.content,
        contentHTML: selectInnerHTMLContentField().contentHTML,
    };

    const { isAllCourses, courseIds } = getCoursesUpsertData(formValues.courses);
    const { isAllGrades, gradeIds } = getGradesUpsertData(formValues.grades);

    const checkedTargetGroup = formValues.targetGroup
        ? [formValues.targetGroup]
        : [UserRoles.USER_GROUP_PARENT, UserRoles.USER_GROUP_STUDENT];

    const receiverIdsList: UpsertNotificationProps["receiverIdsList"] = formValues.students.map(
        (student: { user_id: ArrayElement<UpsertNotificationProps["receiverIdsList"]> }) =>
            student.user_id
    );

    const scheduledAt: UpsertNotificationProps["scheduledAt"] = getScheduleUpsertData({
        scheduleDate: formValues.scheduleDate,
        scheduleTime: formValues.scheduleTime,
        status: formValues.status,
    });

    const questionnaire: UpsertNotificationProps["questionnaire"] = getQuestionnaireUpsertData({
        expirationDate: formValues.expirationDate,
        expirationTime: formValues.expirationTime,
        isAllowResubmission: formValues.isAllowResubmission,
        questionFieldArrayItem: formValues.questionFieldArrayItem,
    });

    const tagIdsList = formValues.tags?.map((tag) => tag.tag_id);

    return {
        notificationId: formValues.notificationId || "",
        status: formValues.status,
        type: KeyNotificationType.NOTIFICATION_TYPE_COMPOSED,
        event: KeyNotificationEvent.NOTIFICATION_EVENT_NONE,
        title: trimmedTitle,
        content: mappedContent,
        gradeIds,
        isAllGrades,
        courseIds,
        isAllCourses,
        receiverIdsList,
        targetGroup: checkedTargetGroup,
        isImportant: Boolean(formValues.isImportantNotification),
        mediaIds: formValues.mediaIds,
        scheduledAt,
        questionnaire,
        tagIdsList,
    };
};

// Because the case cantSchedulePastTimeNotification:

// Error message when upsert scheduled notification with invalid scheduled time
// scheduled time is in the past

// => Should not close the dialog to keep user data
export function checkErrorIfShouldCloseComposeDialog(error: Error) {
    return !Boolean(
        error.message.match("ra.manabie-error.specified.cantSchedulePastTimeNotification")
    );
}

/**
 * @param arr array of string(max length = 26)
 * @param charInMiddle character between array items
 * @description convert from array string into array with alphabet, example: ["2022","Manabie"] => ["A. 2022","B. Manabie"]
 * @returns array of string, begin by alphabet
 */
export function addAlphabetToArrayString(
    arr: Array<string>,
    charInMiddle: "." | "/" | "," = "."
): Array<string> {
    return arr.map(
        (item, index) => `${getAlphabetArray("uppercase")[index]}${charInMiddle} ${item}`
    );
}

export function getDateBasedOnCheckSchedule(
    isSchedule: boolean,
    dateValue: Date | undefined | null
) {
    return isSchedule ? dateValue : new Date();
}

export function checkParamsAndConvertToDate(
    params: string | number | undefined,
    defaultValue: Date | undefined = undefined
): Date | undefined {
    return params ? new Date(params) : defaultValue;
}

/**
 * @param durations durations
 * @param durationUnit duration unit
 * @returns Date() after days, calculate from now.
 */
export const getDateAfterDuration = (
    durations: number,
    durationUnit: keyof DurationLikeObject = "days"
): Date => {
    return DateTime.local()
        .plus({ [durationUnit]: durations })
        .toJSDate();
};

/**
 * @param timeValue timeValue example: 10:00
 * @returns Date with new hour
 */
export const getCurrentDateWithSpecificHour = (timeValue: string) => {
    const [hour, minute] = timeValue.split(":").map(Number);
    return DateTime.local().set({ hour, minute }).toJSDate();
};

/**
 * Convert to "Paging" proto with only limit and offset
 *
 * @param paging paging object {limit: number; offsetInteger: number; offsetString: string; ...}
 * @returns proto class "Paging"
 */
export const toBasicPagingProto = (paging: Paging.AsObject) => {
    const result = new Paging();

    result.setLimit(paging.limit);

    // We only set the offset by one of 2 things, string or integer
    if (Boolean(paging.offsetString)) result.setOffsetString(paging.offsetString);
    else result.setOffsetInteger(paging.offsetInteger);

    return result;
};

export const covertReactNodeToString = (title: React.ReactNode) =>
    typeof title === "string" ? title : "";

/**
 * Convert "Filter default form value" to "Filter field object" with translation
 *
 * @param filterDefaultValues Form filter value by default
 * @returns object after convert with proper format & translation
 */
export const convertFilterFieldsObjects = <T>(
    tNotification: UseResourceTranslateReturn,
    filterDefaultValues: T
): FilterAppliedObjectsMap<T> => {
    const results = Object.keys(filterDefaultValues).map((key) => {
        return [
            key,
            {
                name: key,
                inputLabel: tNotification(`filters.${key}`),
                isApplied: false,
                defaultValue: filterDefaultValues[key],
            },
        ];
    });
    return Object.fromEntries(results);
};

/**
 * @param arr array of object
 * @param
 *      key: props of object wi
 *      charInMiddle character between array items
 * @description convert from array string into array with alphabet, example: ["Tag Name A","Tag Name B"] => ["Tag Name A","Tag Name B"]
 * @returns array of string with string join
 */
export function joinArrayObjectString<T>(
    arr: T[],
    key: keyof T,
    charInMiddle: "." | "/" | "," = ","
): string {
    return arr.map((item: T) => item[String(key)]).join(`${charInMiddle} `);
}
