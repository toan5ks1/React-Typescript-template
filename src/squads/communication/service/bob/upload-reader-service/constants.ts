import { MAX_FILE_SIZE } from "src/common/constants/file-size";

export const maxFileSize = MAX_FILE_SIZE;
