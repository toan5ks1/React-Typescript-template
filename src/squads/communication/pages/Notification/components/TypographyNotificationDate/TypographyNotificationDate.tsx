import { DateTime } from "luxon";
import { KeyNotificationStatus } from "src/common/constants/const";
import { ERPModules, FormatDateOptions } from "src/common/constants/enum";
import { formatDate } from "src/common/utils/time";
import { NotificationStatusKeys } from "src/squads/communication/typings/remote";

import DoubleDash from "src/components/DoubleDash";
import TypographyBase, { TypographyBaseProps } from "src/components/Typographys/TypographyBase";

import useResourceTranslate from "src/squads/communication/hooks/useResourceTranslate";

interface TypographyNotificationDateProps extends TypographyBaseProps {
    date: Date | string | null;
    format?: FormatDateOptions;
    status?: NotificationStatusKeys;
}

const TypographyNotificationDate = (props: TypographyNotificationDateProps) => {
    const { status, date, format = "yyyy/LL/dd, HH:mm", ...rest } = props;

    const tNotification = useResourceTranslate(ERPModules.NOTIFICATIONS);

    const getTextDateOfNotification = (
        date: TypographyNotificationDateProps["date"],
        status?: TypographyNotificationDateProps["status"]
    ): string | JSX.Element => {
        if (status === KeyNotificationStatus.NOTIFICATION_STATUS_SCHEDULED) {
            return tNotification("notificationStatus.NOTIFICATION_STATUS_SCHEDULED");
        }

        // Because when we pass string type to fromJSDate function, it will return invalid date.
        // So we have to initialize Date type to pass to fromJSDate function
        return date && DateTime.fromJSDate(new Date(date)).isValid ? (
            formatDate(date, format)
        ) : (
            <DoubleDash />
        );
    };

    return (
        <TypographyBase data-testid="TypographyNotificationDate__root" variant="body2" {...rest}>
            {getTextDateOfNotification(date, status)}
        </TypographyBase>
    );
};

export default TypographyNotificationDate;
