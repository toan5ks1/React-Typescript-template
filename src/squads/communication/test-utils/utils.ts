import { DateTime } from "luxon";
import { dateIsAfter, formatDate } from "src/common/utils/time";

import { getDateAfterDuration } from "./../common/utils/utils";

import { fireEvent, getByRole, screen, waitFor, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

/**
 * check the dialog open when clicking action button (search, add button)
 * @param testIdTable test id of table
 * @param dialogTitle the title of dialog
 * @param testIdButton test id of button
 */
export const checkOpenDialogWhenClickActionButton = (
    testIdTable: string,
    dialogTitle: string,
    testIdButton: string
) => {
    const openDialogButton = within(screen.getByTestId(testIdTable)).getByTestId(
        testIdButton
    ) as HTMLButtonElement;
    userEvent.click(openDialogButton);
    const actionDialog = screen.getByTestId("DialogWithHeaderFooter_wrapper");

    expect(actionDialog).toBeInTheDocument();
    expect(screen.getByTitle(dialogTitle)).toBeInTheDocument();

    const cancelDialogButton = within(actionDialog).getByTestId(
        "FooterDialogConfirm__buttonClose"
    ) as HTMLButtonElement;
    // close dialog
    userEvent.click(cancelDialogButton);
};

/**
 * check the dialog close when click cancel button or close icon
 * @param testIdButton test id of button
 * @param onClose the onClose function
 */
export const checkCloseUpsertDialog = (testIdButton: string, onClose: () => void) => {
    const closeDialogButton = screen.getByTestId(testIdButton);
    userEvent.click(closeDialogButton);
    const dialogConfirm = screen.getByTestId("DialogWithHeaderFooter_wrapper");
    expect(dialogConfirm).toBeInTheDocument();

    const saveConfirmButton = within(dialogConfirm).getByTestId(
        "FooterDialogConfirm__buttonSave"
    ) as HTMLButtonElement;
    userEvent.click(saveConfirmButton);
    expect(onClose).toBeCalled();
};

/**
 * check error message in form
 * @param numberOfErrorMessage the number of error message
 * @param errorMessage the error message
 */
export const checkErrorMessage = async (numberOfErrorMessage: number, errorMessage: string) => {
    await waitFor(() => {
        const allErrorText = screen.queryAllByText(errorMessage);
        expect(allErrorText).toHaveLength(numberOfErrorMessage);
        allErrorText.forEach((el) => {
            expect(el).toBeInTheDocument();
        });
    });
};

/**
 * change value of text input
 * @param testIdTextInput test id of text input
 * @param textInputValue value of text input
 */
export const changeTextInput = (testIdTextInput: string, textInputValue: string) => {
    const textInput = screen.getByTestId(testIdTextInput) as HTMLInputElement;
    userEvent.type(textInput, textInputValue);
};

/**
 * change value of auto complete input
 * @param autocompleteTestId data test id of autocomplete input
 * @param autocompleteValue value of autocomplete input
 */
export const changeAutocompleteValue = (autocompleteTestId: string, autocompleteValue: string) => {
    const autocompleteInput = within(screen.getByTestId(autocompleteTestId)).getByRole("combobox", {
        hidden: true,
    });
    userEvent.type(autocompleteInput, autocompleteValue);
    userEvent.keyboard("{ArrowDown}");
    userEvent.keyboard("{Enter}");
};

/**
 * change value of editor content
 * @param testIdEditor test id of editor
 * @param editorContent the value of editor content
 */
export const changeEditorContent = (testIdEditor: string, editorContent: string) => {
    const editor = screen.getByTestId(testIdEditor);

    fireEvent.paste(editor, {
        clipboardData: {
            types: ["text/plain"],
            getData: () => editorContent,
        },
    });
};

/**
 * change value of the select component
 * @param selectContainerTestId test id of the select component
 * @param value the value you want to change to in the dropdown
 */
export const changeSelectValue = (selectContainerTestId: string, value: string) => {
    const selectContainer = screen.getByTestId(selectContainerTestId);

    const selectElement = getByRole(selectContainer, "button");
    userEvent.click(selectElement);

    const selectChoice = screen.getByText(value);
    userEvent.click(selectChoice);
};

/**
 * change value of the date picker
 * @param datePickerTestId test id of the date picker
 * @param datePickerValue value of the date picker
 */
export const changeDatePicker = (
    datePickerTestId: string,
    datePickerValue: Date,
    defaultDatePickerValue: Date = getDateAfterDuration(0)
) => {
    const dateInput = within(screen.getByTestId(datePickerTestId)).getByTestId(
        "DatePickerHF__input"
    );

    userEvent.click(dateInput);

    if (dateIsAfter(datePickerValue, defaultDatePickerValue, "month")) {
        const nextMonthButton = screen.getByRole("button", {
            name: /next month/i,
        });
        userEvent.click(nextMonthButton);
    } else if (dateIsAfter(defaultDatePickerValue, datePickerValue, "month")) {
        const previousMonthButton = screen.getByRole("button", { name: /previous month/i });
        userEvent.click(previousMonthButton);
    }

    const convertedDate = DateTime.fromJSDate(datePickerValue).toLocaleString(DateTime.DATE_MED);
    const chosenDate = screen.getByRole("button", {
        name: convertedDate,
    });

    userEvent.click(chosenDate);
    userEvent.click(screen.getByRole("button", { name: /ok/i }));

    expect(dateInput).toHaveValue(formatDate(datePickerValue, "yyyy/LL/dd"));
};
