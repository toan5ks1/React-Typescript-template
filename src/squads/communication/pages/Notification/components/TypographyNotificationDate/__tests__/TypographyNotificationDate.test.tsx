import { KeyNotificationStatus } from "src/common/constants/const";
import { formatDate } from "src/common/utils/time";

import TypographyNotificationDate from "../TypographyNotificationDate";

import { render, RenderResult, screen } from "@testing-library/react";

jest.mock("src/squads/communication/hooks/useResourceTranslate", () => ({
    __esModule: true,
    default: () => (translateKey: string) => `resources.notifications.${translateKey}`,
}));

describe("<TypographyNotificationDate/> ", () => {
    let wrapper: RenderResult;

    const testDate = "2020-01-01T07:00:00.000Z";

    it("should render snapshot", () => {
        wrapper = render(<TypographyNotificationDate date={testDate} />);
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should match value with default formatDate when date is valid", () => {
        wrapper = render(<TypographyNotificationDate date={testDate} />);
        expect(wrapper.getByTestId("TypographyNotificationDate__root")).toHaveTextContent(
            formatDate(testDate, "yyyy/LL/dd, HH:mm")
        );
    });

    it("should match value with formatDate when date is valid and status is sent", () => {
        wrapper = render(
            <TypographyNotificationDate
                date={testDate}
                format="yyyy/LL/dd"
                status={KeyNotificationStatus.NOTIFICATION_STATUS_SENT}
            />
        );
        expect(wrapper.getByTestId("TypographyNotificationDate__root")).toHaveTextContent(
            formatDate(testDate, "yyyy/LL/dd")
        );
    });

    it("should match value when status is draft", () => {
        wrapper = render(
            <TypographyNotificationDate
                date={null}
                status={KeyNotificationStatus.NOTIFICATION_STATUS_DRAFT}
            />
        );
        expect(wrapper.container).toMatchSnapshot();
        expect(wrapper.getByTestId("TypographyNotificationDate__root")).toHaveTextContent("--");
    });

    it("should match value when status is scheduled and column is sentDate", () => {
        wrapper = render(
            <TypographyNotificationDate
                date={testDate}
                status={KeyNotificationStatus.NOTIFICATION_STATUS_SCHEDULED}
            />
        );
        expect(wrapper.container).toMatchSnapshot();
        expect(screen.getByTestId("TypographyNotificationDate__root")).toHaveTextContent(
            "resources.notifications.notificationStatus.NOTIFICATION_STATUS_SCHEDULED"
        );
    });

    it("should display double dash when pass invalid date", () => {
        const invalidDate = "2020-13-31T07:00:00.000Z";
        wrapper = render(
            <TypographyNotificationDate
                date={invalidDate}
                status={KeyNotificationStatus.NOTIFICATION_STATUS_DRAFT}
            />
        );

        expect(wrapper.getByTestId("TypographyNotificationDate__root")).toHaveTextContent("--");
    });
});
