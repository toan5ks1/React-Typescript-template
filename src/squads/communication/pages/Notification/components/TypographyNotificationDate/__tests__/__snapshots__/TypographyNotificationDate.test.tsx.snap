// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypographyNotificationDate/>  should match value when status is draft 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body2 emotion-0"
    data-testid="TypographyNotificationDate__root"
  >
    --
  </p>
</div>
`;

exports[`<TypographyNotificationDate/>  should match value when status is scheduled and column is sentDate 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body2 emotion-0"
    data-testid="TypographyNotificationDate__root"
  >
    resources.notifications.notificationStatus.NOTIFICATION_STATUS_SCHEDULED
  </p>
</div>
`;

exports[`<TypographyNotificationDate/>  should render snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body2 emotion-0"
    data-testid="TypographyNotificationDate__root"
  >
    2020/01/01, 07:00
  </p>
</div>
`;
